backend/pkg/api/connect/service/quota/v1/service.go

```diff
@@ -0,0 +1,169 @@
+var _ dataplanev1connect.QuotaServiceHandler = (*Service)(nil)
+
+// Service implements the handlers for Quota endpoints.
+type Service struct {
+	cfg        *config.Config
+	logger     *slog.Logger
+	consoleSvc console.Servicer
+
+	kafkaClientMapper *kafkaClientMapper
+}
+
+// NewService creates a new Quota service handler.
+func NewService(cfg *config.Config,
+	logger *slog.Logger,
+	consoleSvc console.Servicer,
+) *Service {
+	return &Service{
+		cfg:               cfg,
+		logger:            logger,
+		consoleSvc:        consoleSvc,
+		kafkaClientMapper: &kafkaClientMapper{},
+	}
+}
+
+// ListQuotas lists client quotas based on filter criteria.
+func (s *Service) ListQuotas(ctx context.Context, req *connect.Request[v1.ListQuotasRequest]) (*connect.Response[v1.ListQuotasResponse], error) {
+	kafkaReq := s.kafkaClientMapper.listQuotasRequestToKafka(req.Msg)
+
+	resp, err := s.consoleSvc.DescribeClientQuotas(ctx, kafkaReq)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to describe quotas: %w", err))
+	}
+
+	// Handle errors from the console service
+	if err = kerr.ErrorForCode(resp.ErrorCode); err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to describe quotas: %s", err))
+	}
+
+	// Map console response to protobuf response
+	quotaEntries, err := s.kafkaClientMapper.quotaItemsToProto(resp.Entries)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to map quota items to proto: %w", err))
+	}
+
+	// Add pagination
+	var nextPageToken string
+	if req.Msg.GetPageSize() > 0 {
+		sort.SliceStable(quotaEntries, func(i, j int) bool {
+			if quotaEntries[i].Entity.EntityType != quotaEntries[j].Entity.EntityType {
+				return quotaEntries[i].Entity.EntityType < quotaEntries[j].Entity.EntityType
+			}
+			return quotaEntries[i].Entity.EntityName < quotaEntries[j].Entity.EntityName
+		})
+		page, token, err := pagination.SliceToPaginatedWithToken(quotaEntries, int(req.Msg.PageSize), req.Msg.GetPageToken(), "quotaEntry", func(x *v1.ListQuotasResponse_QuotaEntry) string {
+			return fmt.Sprintf("%d_%s", x.Entity.EntityType, x.Entity.EntityName)
+		})
+		if err != nil {
+			return nil, connect.NewError(
+				connect.CodeInternal,
+				fmt.Errorf("failed to apply pagination: %w", err),
+			)
+		}
+		quotaEntries = page
+		nextPageToken = token
+	}
+
+	response := &v1.ListQuotasResponse{
+		Quotas:    quotaEntries,
+		PageToken: nextPageToken,
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// BatchSetQuota creates client throughput quotas for specified entities.
+func (s *Service) BatchSetQuota(ctx context.Context, req *connect.Request[v1.BatchSetQuotaRequest]) (*connect.Response[v1.BatchSetQuotaResponse], error) {
+	kafkaReq, err := s.kafkaClientMapper.batchSetQuotaRequestToKafka(req.Msg)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	_, err = s.consoleSvc.AlterClientQuotas(ctx, kafkaReq)
```

Comment:
In the single SetQuota you checked for an error (`if err = kerr.ErrorForCode(resp.ErrorCode); err != nil {`), we don't want to error the response here if one of the quotas failed, we must add it as part of our response.

So in the proto `message BatchSetQuotaResponse {}` we have to mimic somewhat what the kafka response does I guess. Otherwise the user has no chance of knowing what quotas were set successfully and which weren't